src/hpr_max_copies.erl

```diff
@@ -0,0 +1,141 @@
+-module(hpr_max_copies).
+
+-export([
+    init/0,
+    check/2,
+    cleanup/1
+]).
+
+-define(ETS, hpr_max_copies_ets).
+-define(MAX_COPIES, max_copies).
+-define(MAX_TOO_LOW, max_copies_max_too_low).
+-define(CLEANUP_TIME, timer:hours(1)).
+
+-spec init() -> ok.
+init() ->
+    %% Table structure
+    %% {Key :: binary(), Counter :: non_neg_integer() :: Timestamp :: integer()}
+    ets:new(?ETS, [
+        public,
+        named_table,
+        set,
+        {write_concurrency, true}
+    ]),
+    ok = scheduled_cleanup(?CLEANUP_TIME),
+    ok.
+
+-spec check(Key :: binary(), Max :: non_neg_integer()) ->
+    ok | {error, ?MAX_TOO_LOW | ?MAX_COPIES}.
+check(_Key, Max) when Max =< 0 ->
```

Comment:
I'm wondering if the function name could be better? To me `check` doesn't imply a state change (counting the checks). Maybe it should be `count`? or `count_with_check`?